File: safety/snip/append.cpp
std::vector<int> append(
        std::vector<int>&& items,
        std::vector<int> const& suffix) {
    items.insert(items.end(), suffix.begin(), suffix.end());
    return items;
}

void test_append() {
    assert((std::vector<int>{1, 2, 3, 4} == append({1, 2}, {3, 4})));
}

File: safety/snip/append.rs
pub fn append(mut items: Vec<i32>, suffix: &[i32]) -> Vec<i32> {
    items.extend_from_slice(suffix);
    items
}

fn test_append() {
    assert_eq!(vec![1, 2, 3, 4], append(vec![1, 2], &[3, 4]));
}
